scipy/signal/_peak_finding.py

```diff
@@ -246,6 +250,107 @@ def argrelextrema(data, comparator, axis=0, order=1, mode='clip'):
     return np.nonzero(results)
 
 
+def _handle_arg_x(value):
```

Comment:
This also increases the readability a lot. I don't need to know what `_check_for_1D` does since I know it is checking the ndim somehow.

Reply:
The reason I prefixed these functions somewhat vaguely with _handle_arg_ is because these function not only validate / assert something but also modify the input argument in which case you'll agree that a return value is needed. Whether throwing these things together is useful is another discussion point which your rightly point out. I'll try to come up with better names but I don't now how to do it otherwise without code duplication.

Reply:
> because these function not only validate / assert something but also modify the input argument in which case you'll agree that a return value is needed. 

Totally agree.